File: frontend/admin/src/api/content.ts
import request from './request'

export interface Article {
  id: number
  title: string
  category: string
  cover_image?: string
  content: string
  status: 'draft' | 'published'
  view_count: number
  published_at?: string
  created_at: string
}

export interface Banner {
  id: number
  title: string
  image_url: string
  link_type: 'none' | 'property' | 'article' | 'url'
  link_value?: string
  position: 'home' | 'property_list'
  sort_order: number
  is_active: boolean
  created_at: string
}

export function getArticles(params: { page?: number; limit?: number; status?: string; keyword?: string }) {
  return request.get('/admin/articles', { params })
}
export function createArticle(data: Partial<Article>) {
  return request.post('/admin/articles', data)
}
export function updateArticle(id: number, data: Partial<Article>) {
  return request.put(`/admin/articles/${id}`, data)
}
export function deleteArticle(id: number) {
  return request.delete(`/admin/articles/${id}`)
}

export function getBanners(params?: { position?: string }) {
  return request.get('/admin/banners', { params })
}
export function createBanner(data: Partial<Banner>) {
  return request.post('/admin/banners', data)
}
export function updateBanner(id: number, data: Partial<Banner>) {
  return request.put(`/admin/banners/${id}`, data)
}
export function deleteBanner(id: number) {
  return request.delete(`/admin/banners/${id}`)
}

File: backend/internal/model/content.go
package model

import "time"

type ArticleCategory string
type ArticleStatus string

const (
	ArticleCategoryNews   ArticleCategory = "news"
	ArticleCategoryPolicy ArticleCategory = "policy"
	ArticleCategoryGuide  ArticleCategory = "guide"
	ArticleCategoryMarket ArticleCategory = "market"

	ArticleStatusDraft     ArticleStatus = "draft"
	ArticleStatusPublished ArticleStatus = "published"
	ArticleStatusOffline   ArticleStatus = "offline"
)

type Article struct {
	ID          uint64          `gorm:"primaryKey;autoIncrement" json:"id"`
	Title       string          `gorm:"size:200;not null" json:"title"`
	Summary     string          `gorm:"size:500;default:''" json:"summary"`
	Content     string          `gorm:"type:longtext;default:''" json:"content,omitempty"`
	CoverImage  string          `gorm:"size:512;default:''" json:"cover_image"`
	Category    ArticleCategory `gorm:"type:enum('news','policy','guide','market');not null;default:'news'" json:"category"`
	Author      string          `gorm:"size:50;default:''" json:"author"`
	ViewCount   uint32          `gorm:"not null;default:0" json:"view_count"`
	Status      ArticleStatus   `gorm:"type:enum('draft','published','offline');not null;default:'draft';index" json:"status"`
	PublishedAt *time.Time      `json:"published_at"`
	CreatedBy   uint64          `gorm:"default:0" json:"created_by"`
	CreatedAt   time.Time       `json:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at"`
}

func (Article) TableName() string { return "articles" }

type BannerPosition string
type BannerLinkType string

const (
	BannerPositionHome BannerPosition = "home"
	BannerPositionList BannerPosition = "list"

	BannerLinkTypeProperty BannerLinkType = "property"
	BannerLinkTypeArticle  BannerLinkType = "article"
	BannerLinkTypeExternal BannerLinkType = "external"
	BannerLinkTypeNone     BannerLinkType = "none"
)

type Banner struct {
	ID        uint64         `gorm:"primaryKey;autoIncrement" json:"id"`
	Title     string         `gorm:"size:100;default:''" json:"title"`
	ImageURL  string         `gorm:"size:512;not null" json:"image_url"`
	LinkType  BannerLinkType `gorm:"type:enum('property','article','external','none');not null;default:'none'" json:"link_type"`
	LinkValue string         `gorm:"size:500;default:''" json:"link_value"`
	Position  BannerPosition `gorm:"type:enum('home','list');not null;default:'home';index" json:"position"`
	SortOrder int            `gorm:"not null;default:0" json:"sort_order"`
	IsActive  bool           `gorm:"not null;default:true" json:"is_active"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
}

func (Banner) TableName() string { return "banners" }

type Area struct {
	ID        uint64 `gorm:"primaryKey;autoIncrement" json:"id"`
	ParentID  uint64 `gorm:"not null;default:0;index" json:"parent_id"`
	Name      string `gorm:"size:50;not null" json:"name"`
	SortOrder int    `gorm:"not null;default:0" json:"sort_order"`
	IsActive  bool   `gorm:"not null;default:true" json:"is_active"`
}

func (Area) TableName() string { return "areas" }
